### PR TITLE
[NEW] REST endpoint to recover forgotten password

### LINKED ISSUE #10315: [NEW] REST API to perform forgot password
Creates a REST API to perform `forgot password`. 
Here is the idea: 

- User calls the forgot password REST API with an email;
- The API checks if the email is valid (I mean, if it exists at least on the DB);
- If the user email is valid the API dispatches an email with a link to able the user to redefine its password (like the web app does).

### PR DESCRIPTION
**Pull request by MarcosSpessatto**
Add REST /forgotPassword endpoint and tests.
Closes #10315.

### CODE REVIEW

File: packages/rocketchat-api/server/api.js [hunk 1]
```diff
@@ -325,6 +325,22 @@ class API extends Restivus {
 			}
 			return response;
 		};
+		const forgotPassword = function() {
```
Comment (graywolf336) [inline packages/rocketchat-api/server/api.js]: Honestly, I'm not sure if this should live in here or in the users file. Thoughts @sampaiodiego

Comment (sampaiodiego) [reply packages/rocketchat-api/server/api.js]: I agree, the endpoint could be something like `/users.forgotPassword`

Comment (graywolf336) [reply packages/rocketchat-api/server/api.js]: My thoughts as well. :+1:

**Comment by filipedelimabrito** [comment]
@MarcosSpessatto @sampaiodiego Is there a doc for this endpoint? Where in case of yes?

**Comment by sampaiodiego** [comment]
Yes! https://rocket.chat/docs/developer-guides/rest-api/users/forgotpassword/

**Pull request merged**